pkg/engine/vmextensions.go

```diff
@@ -234,39 +234,53 @@ func CreateAgentVMASAKSBillingExtension(cs *api.ContainerService, profile *api.A
 func CreateCustomExtensions(properties *api.Properties) []DeploymentARM {
 	var extensionsARM []DeploymentARM
 
-	for _, extensionProfile := range properties.ExtensionProfiles {
-		if properties.MasterProfile != nil {
+	if properties.MasterProfile != nil {
+		// The first extension needs to depend on the billing extension created for all pools
+		// Each proceeding extension needs to depend on the previous one to avoid ARM conflicts in the Compute RP
+		nextDependsOn := "[concat('Microsoft.Compute/virtualMachines/', variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')), '/extensions/computeAksLinuxBilling')]"
```

Comment:
doesn't the billing extension change names depending on AKS/AKS Engine? Also it can be disabled. I don't think we can hardcode a dependency on `computeAksLinuxBilling` here.

Reply:
the name of the extension doesn't change (which is weird)

Reply:
I didn't know that the billing extension can be disabled, is there a way to detect that in the properties @CecileRobertMichon.

Reply:
You could disable the extension though. See `IsAKSBillingEnabled`.

Reply:
Thanks, I'll take a look

Reply:
Yeah looks like it's the type that changes, not the name. AKS did change the vmss one at some point https://github.com/Azure/aks-engine/pull/1561/files but that was for VMSS so shouldn't affect this.

Reply:
It looks like we only need to serialize the deployments of the 'CustomScriptExtension' extensions. The billing extensions are their own type of extension so we don't need to worry about those.

It also looks like the way aks-engine is using custom script extensions for additional extensions is a bit fragile. According to https://azure.microsoft.com/en-us/blog/automate-linux-vm-customization-tasks-using-customscript-extension/ you are supposed to refernce all of the scripts you want to run in a single ARM resource since there can only be on 'CustomScriptExtension' active on a machine/scale set at any given time. 

A larger refactor should probably be considered when we have time and/or if we need to support additional extensions for scale sets.

This is working pretty reliably as is in my local testing.